package/MDAnalysis/coordinates/chemfiles.py

```diff
@@ -52,6 +53,14 @@
 else:
     HAS_CHEMFILES = True
 
+# Allow building documentation even if chemfiles is not installed
+if not HAS_CHEMFILES and 'sphinx' in sys.modules:
+    import imp
+    class MockTrajectory: pass
+    chemfiles = imp.new_module("chemfiles")
+    chemfiles.Trajectory = MockTrajectory
+    HAS_CHEMFILES = True
+
```

Comment:
That's a workaround but a bit ugly in that it adds code to the library that has nothing to do with its purpose, namely generation of documentation. Additionally, this brings down coverage.

I was thinking along https://docs.python.org/3/library/unittest.mock.html and add a MagicMock to the sphinx conf.py, something like
```python
try:
   import chemfiles
except ImportError:
   # mock it
   import sys
   from unittest.mock import MagicMock
   sys.modules['chemfiles'] = MagicMock()
```
The idea would be to have the (mocked) module in `sys.path` already so when mda is trying to `import chemfiles` it will just take the one from `sys.path`.

I am not 100% sure that this works. @richardjgowers has done some of these shenanigans in the past.

Reply:
Could you just move the `if HAS_CHEMFILES` into `__init__`?

Reply:
I changed the code so that it will mock `chemfiles` as long as we it cannot find one (regardless we are building a doc or not). So now we don't need to add `HAS_CHEMFILES` as a requirement for the existence of `ChemfilesPicklable`.
 
With that the test can pass, and the doc can be built with/without `chemfiles`.